src/Kestrel.Core/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -549,5 +516,14 @@ private sealed class UTF8EncodingSealed : UTF8Encoding
 
             public override byte[] GetPreamble() => Array.Empty<byte>();
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal unsafe static Span<byte> UnsafeAsSpan(this ref ReadOnlySpan<byte> span)
+        {
+            fixed (byte* b = span)
+            {
+                return new Span<byte>(b, span.Length);
```

Comment:
Is this valid? The returned Span is now pointing at an untracked pointer and the data under it could be moved? (Especially for slow `Span<T>` as it would now have no rooting `object`)

/cc @jkotas

Slightly ironically you could do this more safely without `fixed` on .NET Core
```csharp
internal static Span<byte> UnsafeAsSpan(this ReadOnlySpan<byte> span)
{
    return MemoryMarshal.CreateSpan(ref MemoryMarshal.GetReference(span), span.Length);
}
```
But that doesn't help with the .NET Standard/slow Span

Reply:
It isn't. I'm going to move this down to where this actually gets modified.

Reply:
Yes this is unsafe 😄

Reply:
It's gone. :)